torch/csrc/autograd/python_cpp_function.cpp

```diff
@@ -251,15 +251,17 @@ static std::unordered_set<PyTypeObject*> cpp_function_types_set;
 struct DefaultFunctionType {
   DefaultFunctionType() : type() {
     _initFunctionPyTypeObject(type, "CppFunction", nullptr, nullptr);
-    Py_INCREF(&type);
```

Comment:
You don't need the incref?

Reply:
Correct, see my comment above: https://github.com/pytorch/pytorch/pull/125875#discussion_r1596088552
tl;dr: now that we properly initialize the header in _initFunctionPyTypeObject (was kept all 0s before), this function sets the refcount to 1 properly.